source/includes/fundamentals/code-snippets/CRUD/sort.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	var uri string
+	if uri = os.Getenv("DRIVER_REF_URI"); uri == "" {
+		log.Fatal("You must set your `MONGODB_URI' environmental variable. See\n\t https://docs.mongodb.com/drivers/go/current/usage-examples/")
+	}
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	client.Database("tea").Collection("ratings").Drop(context.TODO())
+
+	// begin insertDocs
+	coll := client.Database("tea").Collection("ratings")
+	docs := []interface{}{
+		bson.D{{"type", "Masala"}, {"rating", 10}},
+		bson.D{{"type", "Assam"}, {"rating", 5}},
+		bson.D{{"type", "Oolong"}, {"rating", 7}},
+		bson.D{{"type", "Earl Grey"}, {"rating", 8}},
+		bson.D{{"type", "English Breakfast"}, {"rating", 5}},
+	}
+
+	result, insertErr := coll.InsertMany(context.TODO(), docs)
+	if insertErr != nil {
+		panic(insertErr)
+	}
+	//end insertDocs
+	fmt.Printf("Number of documents inserted: %d\n", len(result.InsertedIDs))
+
+	fmt.Println("Ascending Sort:")
+	//begin ascending sort
+	ascendingFilter := bson.D{{}}
+	acendingSort := bson.D{{"rating", 1}}
+	ascendingOptions := options.Find().SetSort(acendingSort)
+
+	ascendingCursor, ascendingErr := coll.Find(context.TODO(), ascendingFilter, ascendingOptions)
+
+	var ascendingResults []bson.M
```

Comment:
```suggestion
	var ascendingResults []bson.D
```
We also sometimes use `[]bson.M` as the `results` parameter for `cursor.All()` in our local examples. I'm not sure why... Unless the order of fields in each document truly doesn't matter, let's just stick with `bson.D`.